cpp/cmake_modules/ThirdpartyToolchain.cmake

```
@@ -1812,12 +1812,12 @@ if(ARROW_WITH_PROTOBUF)
   else()
     set(ARROW_PROTOBUF_REQUIRED_VERSION "2.6.1")
   endif()
-  if(ARROW_FLIGHT)
-    set(ARROW_PROTOBUF_ARROW_CMAKE_PACKAGE_NAME "ArrowFlight")
-    set(ARROW_PROTOBUF_ARROW_PC_PACKAGE_NAME "arrow-flight")
-  else()
+  if(ARROW_ORC OR ARROW_WITH_OPENTELEMETRY)
     set(ARROW_PROTOBUF_ARROW_CMAKE_PACKAGE_NAME "Arrow")
     set(ARROW_PROTOBUF_ARROW_PC_PACKAGE_NAME "arrow")
+  elseif(ARROW_FLIGHT)
```

Can we simplify this?

```suggestion
  else()
```

---

I think that is too much implicit logic.
In theory it should even be a list of `ARROW_PROTOBUF_ARROW_PC_PACKAGE_NAMES`, but in practice it is not a problem because `arrow-flight` links to `arrow`.

---

This is a good idea. Added.

---

Thanks. Could you fix lint error?